auth_oidc/controllers/main.py

```diff
@@ -48,3 +51,26 @@ def list_providers(self):
                     provider["auth_endpoint"], url_encode(params)
                 )
         return providers
+
+
+class OpenIDLogout(Session):
+
+    @http.route("/web/session/logout", type="http", auth="none")
+    def logout(self, redirect="/web/login"):
+        user = request.env["res.users"].sudo().browse(request.session.uid)
+        if user.oauth_provider_id.id:
+            p = request.env["auth.oauth.provider"].sudo().browse(user.oauth_provider_id.id)
+            if p.logout_endpoint:
+                r = request.env['ir.config_parameter'].sudo().get_param('web.base.url') + redirect
```

Comment:
@ap-wtioit is redirect expected to always be relative to the base URL? If full URLs are supported then line 64 above would need to be updated to be optional.

Reply:
i think only relative redirects are used in odoo but there is no technical limit for that and the redirect paramter is sent directly to werkzeug: https://github.com/odoo/odoo/blob/14.0/addons/web/controllers/main.py#L1319

so yes that's a good catch and web.base.url should only be used when the url is not starting with a protocol (i guess http(s) will suffice), also there is the possibility that the redirect is relative to /web/session/logout (e.g. `../login`, or `destroy`) but i think this is also not used currently (and i tested it only locally for checking security)

Reply:
i can only see `destroy` being used in one Odoo Enterprise application "project_timesheet_synchro / Synchronization with the external timesheet application" and that code upon login does not support the things added by auth_oidc so i would skip this usecase here.

Reply:
So is this good to go?

Reply:
I'd like to test it again (most likely tomorrow) but yes this looks good to me now.